cpp/include/raft/neighbors/detail/nn_descent.cuh

```diff
@@ -1410,10 +1430,21 @@ void build(raft::resources const& res,
                            .node_degree           = extended_graph_degree,
                            .internal_node_degree  = extended_intermediate_degree,
                            .max_iterations        = params.max_iterations,
-                           .termination_threshold = params.termination_threshold};
+                           .termination_threshold = params.termination_threshold,
+                           .output_graph_degree   = params.graph_degree};
 
   GNND<const T, int> nnd(res, build_config);
-  nnd.build(dataset.data_handle(), dataset.extent(0), int_graph.data_handle());
+
+  if (idx.distances().has_value()) {
+    nnd.build(dataset.data_handle(),
+              dataset.extent(0),
+              int_graph.data_handle(),
+              params.return_distances,
+              idx.distances().value().data_handle());
+  } else {
+    RAFT_EXPECTS(
+      false, "Distance view not allocated. Using NN Descent requires return_distances to be true");
```

Comment:
The else condition should only fail if user wants us to return distances but has not allocated memory, right?
```suggestion
    RAFT_EXPECTS(params.return_distances, "Distance view not allocated. Using NN Descent requires return_distances to be true");
```

Reply:
you are right!